Primo.js

```diff
@@ -363,37 +358,184 @@ function importPNX(text) {
 	item.callNumber = ZU.xpathText(doc, '//enrichment/classificationlcc');
 	
 	item.complete();
+}
+
+function cleanAuthor(str) {
+	return str
+		// Remove year
+		.replace(/\s*,?\s*\(?\d{4}-?(\d{4})?\)?/g, '')
+		// Remove things like (illustrator). TODO: use this to assign creator type?
+		.replace(/\s*,?\s*\([^()]*\)$/, '');
+}
+
+function fetchCreators(item, creators, type, splitGuidance) {
+	for(var i=0; i<creators.length; i++) {
+		var creator = ZU.unescapeHTML(creators[i].textContent).split(/\s*;\s*/);
+		for(var j=0; j<creator.length; j++) {
+			var c = cleanAuthor(creator[j]);
+			c = ZU.cleanAuthor(
+				splitGuidance[c.toLowerCase()] || c,
+				type,
+				true
+			);
+			
+			if (!c.firstName) {
+				delete c.firstName;
+				c.fieldMode = 1;
+			}
+			
+			item.creators.push(c);
+		}
+	}
+}
+
+function extractNumPages(str) {
+	// Borrowed from Library Catalog (PICA). See #756
+	//make sure things like 2 partition don't match, but 2 p at the end of the field do
+	// f., p., and S. are "pages" in various languages
+	// For multi-volume works, we expect formats like:
+	//   x-109 p., 510 p. and X, 106 S.; 123 S.
+	var numPagesRE = /\[?((?:[ivxlcdm\d]+[ ,\-]*)+)\]?\s+[fps]\b/ig,
```

Comment:
I suggest to add a `\b` in the beginning of this regexp, i.e.
```
var numPagesRE = /\b\[?((?:[ivxlcdm\d]+[ ,\-]*)+)\]?\s+[fps]\b/ig,
```
Currently it will match (highlighted bold) in the string "an<b>d x, 106 s.</b>"